### PR TITLE
Assignment :: Hooks and Routes

### PR DESCRIPTION
**Pull request by ginabak**
# TL;DR

Integrated functionality with ReactRouter & ApolloProvider! (Backend & Frontend are now connected, yay!) 

### Quick Preview (since you won't be able to `npm start` this repo) 
Screen Shot 2020-03-27 at 10 20 53 PM

### (that data is from RAILS, so fun!) 🚀

# What does this PR do? 

This PR fulfills this checklist: 

- [x] Convert every single class into a functional component
- [x] Implement ReactRouter into your blog (corrected - since it was implemented for last assignment) 
- [x] Have 3 Routes setup through ReactRouter that have different layouts or uses, see `app.js` for all 3 Routes (will implement more later). 
- [x] Capable of connecting with backend, specifically with this rails project (when `dev server`): 'https://rails-data-project.myshopify.io/graphql'

### CODE REVIEW

File: gina-blog-app/src/components/BlogPost/BlogPost.js [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+        <Link to={'post/'+ post.id} className='title-name'>{post.title}</Link>
+      </h1>
+      <h4>
+        by 
+      <Link to='/' className='author-name'> {post.user.firstName}</Link>
+      </h4>
+      <p>{post.content.substring(0,25)} . . . </p>
       </div>
+      {/* <Comments comment={post}/> */}
     </div>
```
Comment (derekshopify) [inline gina-blog-app/src/components/BlogPost/BlogPost.js:31]: Can we make some of these div's more semantic? Articles? Sections?

File: gina-blog-app/src/components/HomePage/HomePage.js [hunk 2]
```diff
@@ -1,31 +1,74 @@
...
+                );
+                })
+              }
+              </div>
+            </div>
+
+          </section>
+        );
+
+      }}
```
Comment (derekshopify) [inline gina-blog-app/src/components/HomePage/HomePage.js:71]: Watch your formatting here.

File: gina-blog-app/src/components/LatestBlogPost/LatestBlogPost.js [hunk 3]
```diff
@@ -1,52 +1,15 @@
...
-          </section>
-        );
-      }}
-    </Query>
+export function LatestBlogPost({latestPost}) {
+  return( 
+    <div className='latest-blog-post-position'>
+      <img className='latest-post-image' src={ plant }></img>
+      <BlogPost post={latestPost}/>
+    </div>
```
Comment (derekshopify) [inline gina-blog-app/src/components/LatestBlogPost/LatestBlogPost.js:12]: Article instead of div?

File: gina-blog-app/src/components/StaticFooter/StaticFooter.js [hunk 4]
```diff
@@ -1,44 +1,42 @@
...
+            <img className='social-media-logo' id='pinterest' src={ require('./images/pinterest.png') } alt='twitter logo'/>
+          </a>
+        </li>
+        <li>
+          <a href='https://medium.com/swlh/fashion-blockchain-ai-oh-my-afc161db3fdf'>
+            <img className='social-media-logo' id='medium' src={ require('./images/medium.png') } alt='twitter logo'/>
+          </a>
+        </li>
+      </ul>
+    </div>
```
Comment (derekshopify) [inline gina-blog-app/src/components/StaticFooter/StaticFooter.js:38]: Is there a more semantic version of div when dealing with footers?
